### PR TITLE
ILM setPriority corrections for a 0 value

### LINKED ISSUE #37652: [CI] SetPriorityActionTests has invalid test for testNonPositivePriority
Zero is a valid value according to the class under test but `testNonPositivePriority` is expecting an exception. 

Build failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/290/console

Reproduce with:

    ./gradlew :x-pack:plugin:core:unitTest -Dtests.seed=9FD1CC17D355FD0D -Dtests.class=org.elasticsearch.xpack.core.indexlifecycle.SetPriorityActionTests -Dtests.method="testNonPositivePriority" -Dtests.security.manager=true -Dtests.jvm.argline="-XX:-UseConcMarkSweepGC -XX:+UseG1GC" -Dtests.locale=es-DO -Dtests.timezone=Asia/Katmandu -Dcompiler.java=11 -Druntime.java=8

### PR DESCRIPTION
**Pull request by jakelandis**
This commit fixes the test case that ensures only a priority
less then 0 is used with testNonPositivePriority. This also
allows the HLRC to support a value of 0.

Closes #37652

### CODE REVIEW

**Comment by jakelandis** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/docs/en/watcher/transform/search.asciidoc [hunk 1]
```diff
@@ -56,10 +56,6 @@ The following table lists all available settings for the search transform:
 
 | `request.indices`                             | no        | all indices       | One or more indices to search on.
 
-| `request.types`                               | no        | all types         | One or more document types to search on (may be a
```
Comment (dakrone) [inline x-pack/docs/en/watcher/transform/search.asciidoc]: I think this doc change snuck into this PR? It seems completely separate from this change

**Review by dakrone — APPROVED** [review]
LGTM, left one comment about the doc change

File: x-pack/docs/en/watcher/transform/search.asciidoc (hunk 1 above)
Comment (dakrone) [reply x-pack/docs/en/watcher/transform/search.asciidoc]: Hah you fixed it right as I commented :)

Comment (jakelandis) [reply x-pack/docs/en/watcher/transform/search.asciidoc]: your fast!  yeah, I was on the wrong branch, that commit reverted.

**Comment by jakelandis** [comment]
@dakrone thanks for the review ! will merge on green.

**Pull request merged**